docs/GroupDefs.hpp

```diff
@@ -168,42 +168,231 @@ initialization for the user that can otherwise lead to difficult to debug
+  as any non-standard options that must be specified in the metavariables and
+  their current values. An example of a help string for one of the testing
+  executables is:
+  \snippet Test_AlgorithmCore.cpp help_string_example
+- `component_list`: a `tmpl::list` of the parallel components (described below)
+  that are to be created. Most evolution executables will have the
+  `DgElementArray` parallel component listed. An example of a `component_list`
+  for one of the test executables is:
+  \snippet Test_AlgorithmCore.cpp component_list_example
+- `Phase`: an `enum class` that must contain at least `Initialization` and
+  `Exit`. Phases are described in the next section.
+- `determine_next_phase`: a static function with the signature
+  \code
+    static Phase determine_next_phase(
+      const Phase& current_phase,
+      const Parallel::CProxy_ConstGlobalCache<EvolutionMetavars>& cache_proxy)
+      noexcept;
+  \endcode
+  What this function does is described below in the discussion of phases.
+
+There are also several optional members:
+
+- `input_file`: a `static constexpr OptionString` that is the default name of
+  the input file that is to be read. This can be overridden at runtime by
+  passing the `--input-file` argument to the executable.
+- `ignore_unrecognized_command_line_options`: a `static constexpr bool` that
+  defaults to `false`. If set to `true` then unrecognized command line options
+  are ignored. Ignoring unrecognized options is generally only necessary for
+  tests where arguments for the testing framework, Catch, are passed to the
+  executable.
+
+### Phases of an Execution
+
+Global synchronization points, where all cores wait for each other are
+undesirable for scalability reasons. However, they are sometimes inevitable for
+algorithmic reasons. That is, in order to actually get a correct solution you
+need to have a global synchronization. SpECTRE executables can have multiple
+phases, where after each phase a global synchronization occurs. By global
+synchronization we mean that no parallel components are executing or have more
+tasks to execute: everything is waiting on a task to perform.
+
+Every executable must have at least two phases, `Initialization` and `Exit`. The
+next phase is decided by the static member function `determine_next_phase` in
+the metavariables. Currently this function has access to the phase that is
+ending, and also the global cache. In the future we will add support for
+receiving data from various components to allow for more complex decision
+making. Here is an example of a `determine_next_phase` function and the `Phase`
+enum class:
+\snippet Test_AlgorithmCore.cpp determine_next_phase_example
+
+Another restriction is that currently all Algorithms (described below) must
+execute the same %Actions (also described below) in all phases. This restriction
+is also planned on being relaxed if the need arises.
```

Comment:
Describe what is special about `Phase::Initialization` and `Phase::Exit`

Reply:
Yeah it would be good to have a bit of an overview of how the executable is launched in the Charm++ infrastructure, i.e. where the `initialize` function on the parallel components are called.

Reply:
It is a bit unclear how the algorithm ties phases and parallel components together I think. You could move this sentence about executing the same actions (perhaps substitute parallel components for actions) into the _Algorithm_ section, and provide a few more details.

Reply:
Okay, I'm moving this to the Algorithm section at the end as a note. I think it'll be clearer what this means after the algorithm is explained, and moving it into a note should indicate that it's not essential to understand that. I added this because Mark had asked about it at some point and want to make sure it's documented so if someone else wants this in the future they at least have some starting point.